mono/metadata/Makefile.am

```diff
@@ -433,6 +437,10 @@ libmonoruntimesgen_la_SOURCES = $(common_sources) $(icall_tables_sources) $(ilge
 libmonoruntimesgen_la_CFLAGS = $(SGEN_DEFINES) @CXX_ADD_CFLAGS@
 libmonoruntimesgen_la_LIBADD = libmonoruntime-config.la libmonoruntime-support.la
 
+libmonoruntimecore_la_SOURCES = $(libmonoruntimesgen_la_SOURCES)
+libmonoruntimecore_la_CFLAGS = $(libmonoruntimesgen_la_CFLAGS) -D ENABLE_NETCORE
```

Comment:
why do we need the explicit `-D ENABLE_NETCORE`? it should already be set by the `AC_DEFINE(ENABLE_NETCORE,..)` in configure.ac when netcore is enabled.

Reply:
Because it's output library-specific define

Reply:
What is the intention here ? Do we want to build a netcore version of the runtime in parallel with the normal version ? That would require compiling everything twice.

Reply:
I guess that what was `--with-core=yes` and `--with-core=only` were about but I think we should have `--with-core=only` and build only netcore output in Mono is configured that way

Reply:
Will update the automake to work with `--with-core=only`